bin/metababel

```diff
@@ -87,30 +98,71 @@ end
 # Indeeed the <%= body %> will be indented, but we don't don't want it,
 # in the body string is already indented
 # But we clean the white space empty line afterward \o/
-def wrote_event_dispatchers(folder, t)
+def wrote_event_dispatchers(folder, t, matching_dispatch_data, callback_types)
   event_name = 'event'
+
   event_class_dispatchers = t.map_event_classes_with_index do |e, index_stream_class, index_event_class, default_clock_class|
     arg_variables = {}
     body = Babeltrace2Gen.context(indent: 1) do
       e.get_getter(event: event_name, arg_variables: arg_variables)
     end
-    EventClassInfo.new(e.name, arg_variables.fetch('outputs', []), body, index_stream_class, index_event_class, default_clock_class,
-                       arg_variables.fetch('outputs_allocated', []))
-  end
 
-  stream_classes_matching_dispatchers = t.stream_classes.map do |s|
-    arg_variables = {}
-    body = ''
-    if s.event_common_context_field_class
-      body = Babeltrace2Gen.context(indent: 1) do
-        s.get_getter(event: event_name, arg_variables: arg_variables)
+    # Automatic event dispatcher
+    evt_args = []
+    evt_args += [GeneratedArg.new('int64_t', '_timestamp')] if default_clock_class
+    evt_args += arg_variables.fetch('outputs',[])
+    dispatchers = [DispatcherInfo.new(e.name, evt_args, 'event')]
+
+    # User event dispatchers
+    dispatchers += matching_dispatch_data.filter {|md| md[:name] if md[:regex].match?(e.name) }.map do |md|
+      dispatcher_evt_args = md.fetch(:args,[]).map do |arg|
+        dispatcher_evt_arg = evt_args.find{ |e| e.name == arg } 
+        raise "User event #{dispatcher_evt_arg} argument not present in model." unless dispatcher_evt_arg
+        dispatcher_evt_arg
       end
+      DispatcherInfo.new(md[:name], dispatcher_evt_args, md[:name])
+    end
+
+    EventClassInfo.new(
+      e.name, arg_variables.fetch('outputs', []), 
+      body, index_stream_class, index_event_class, 
+      default_clock_class,
+      arg_variables.fetch('outputs_allocated', []),
+      dispatchers
+    )
+  end
+
+  # Registers for automatic events
+  register_callbacks = event_class_dispatchers.map do |event|
+    dispatcher = event.dispatchers.find { |d| d.type == 'event' }
```

Comment:
why you need to filter here and bellow?

Reply:
You are right we do not nee to filter at all !!!